insonmnia/worker/server.go

```diff
@@ -1096,6 +1096,70 @@ func (m *Worker) setupRunningContainers() error {
 	return nil
 }
 
+func (m *Worker) tasksGC() error {
+	tk := time.NewTicker(m.cfg.TasksGCPeriod)
+	for {
+		select {
+		case <-m.ctx.Done():
+			return errors.New("context cancelled")
+		case <-tk.C:
+			log.G(m.ctx).Debug("going to GC containers")
+			if err := m.removeExpiredContainers(); err != nil {
+				log.G(m.ctx).Warn("failed to GC containers", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (m *Worker) removeExpiredContainers() error {
+	dockerClient, err := client.NewEnvClient()
+	if err != nil {
+		return err
+	}
+	defer dockerClient.Close()
+
+	// Collect only running containers.
+	containers, err := dockerClient.ContainerList(m.ctx, types.ContainerListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, container := range containers {
+		if _, ok := container.Labels[overseerTag]; ok {
+			dealID, ok := container.Labels[dealIDTag]
+			if !ok {
+				log.G(m.ctx).Warn("container has no deal id info", zap.String("container_id", container.ID))
+				continue
+			}
+
+			bigDealID, ok := big.NewInt(0).SetString(dealID, 10)
+			if !ok {
+				log.G(m.ctx).Warn("container has corrupted deal id",
+					zap.String("container_id", container.ID), zap.Any("deal_id", dealID))
+				continue
+			}
+
+			dealInfo, err := m.getDealInfo(sonm.NewBigInt(bigDealID))
```

Comment:
why don't you check if worker know about this deal instead of going to blockchain?
Also in your case you need to check if this is our deal

Reply:
wtf, `getDealInfo()` does not go to blockchain